src/isotools/_transcriptome_io.py

```diff
@@ -411,41 +429,60 @@ def add_sample_from_bam(self, fn, sample_name=None, barcode_file=None, fuzzy_jun
                     #        if qual:
                     #            tr['mutations'][sample_name][pos][alt][1].append(qual) #assuming the quality already accounts for cov>1
 
+                    # checks if there is any "clipping" in the alignment, which occurs when part of the read doesn't align to the reference genome
+                    # CIGAR string in a SAM/BAM file represents the operations performed on each aligned segment of the read:
+                    #   '4' corresponds to "soft clipping," which means that the aligned segment is present in the SEQ but does not align to the reference
                     if 4 in read.cigartuples:  # clipping
                         s_name = sample_name if not barcodes else barcodes[tags['XC']]
                         clip = get_clipping(read.cigartuples, read.reference_start)
                         tr.setdefault('clipping', {}).setdefault(s_name, {}).setdefault(clip, 0)
                         tr['clipping'][s_name][clip] += cov
+                
                 for tr_interval in transcripts:
                     tr = tr_interval.data
                     tr_ranges = tr.pop('range')
                     # tr_ranges=tr['range']
+
+                    # get start and end positions of the transcript, and update the starts and ends dictionaries with coverage information
```

Comment:
This is not just the transcript, but `r` loops over each exon of the transcript, if I understand it correctly.